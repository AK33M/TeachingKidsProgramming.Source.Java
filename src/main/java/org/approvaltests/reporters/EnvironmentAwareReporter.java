package org.approvaltests.reporters;

import org.teachingkidsprogramming.approvals.lite.ApprovalFailureReporter;

public interface EnvironmentAwareReporter extends ApprovalFailureReporter
{
  boolean isWorkingInThisEnvironment(String forFile);
}
