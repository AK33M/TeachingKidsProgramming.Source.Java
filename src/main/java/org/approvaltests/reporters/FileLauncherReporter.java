package org.approvaltests.reporters;


import org.approvaltests.core.ApprovalFailureReporter;

import org.teachingkidsprogramming.util.TestUtils;
import com.spun.util.io.FileUtils;

public class FileLauncherReporter implements ApprovalFailureReporter
{
  //@Override
  public void report(String received, String approved) throws Exception
  {
    if (FileUtils.isNonEmptyFile(approved))
    {
      TestUtils.displayFile(approved);
    }
    TestUtils.displayFile(received);
  }
}
