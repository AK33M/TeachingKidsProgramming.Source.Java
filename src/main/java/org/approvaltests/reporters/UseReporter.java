package org.approvaltests.reporters;

import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;

import org.teachingkidsprogramming.approvals.lite.ApprovalFailureReporter;

@Retention(RetentionPolicy.RUNTIME)
public @interface UseReporter {
  Class<? extends ApprovalFailureReporter>[] value();
}
