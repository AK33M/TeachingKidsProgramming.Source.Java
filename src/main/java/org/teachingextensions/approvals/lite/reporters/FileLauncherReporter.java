package org.teachingextensions.approvals.lite.reporters;


import org.teachingextensions.approvals.lite.ApprovalFailureReporter;

import org.teachingkidsprogramming.util.TestUtils;
import org.teachingkidsprogramming.util.io.FileUtils;

public class FileLauncherReporter implements ApprovalFailureReporter
{
  //@Override
  public void report(String received, String approved) throws Exception
  {
    if (FileUtils.isNonEmptyFile(approved))
    {
      TestUtils.displayFile(approved);
    }
    TestUtils.displayFile(received);
  }
}
