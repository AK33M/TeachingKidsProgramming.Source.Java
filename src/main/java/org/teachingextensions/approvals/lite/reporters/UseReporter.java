package org.teachingextensions.approvals.lite.reporters;

import org.teachingextensions.approvals.lite.ApprovalFailureReporter;

import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;

@Retention(RetentionPolicy.RUNTIME)
public @interface UseReporter {
    Class<? extends ApprovalFailureReporter>[] value();
}
