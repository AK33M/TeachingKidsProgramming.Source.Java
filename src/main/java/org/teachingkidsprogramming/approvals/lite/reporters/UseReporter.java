package org.teachingkidsprogramming.approvals.lite.reporters;

import org.teachingkidsprogramming.approvals.lite.ApprovalFailureReporter;

import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;

@Retention(RetentionPolicy.RUNTIME)
public @interface UseReporter {
    Class<? extends ApprovalFailureReporter>[] value();
}
