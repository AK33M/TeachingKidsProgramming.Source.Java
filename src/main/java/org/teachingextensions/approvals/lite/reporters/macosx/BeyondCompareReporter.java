package org.teachingextensions.approvals.lite.reporters.macosx;

import org.teachingextensions.approvals.lite.reporters.GenericDiffReporter;

import java.text.MessageFormat;

public class BeyondCompareReporter extends GenericDiffReporter {
  public static final BeyondCompareReporter INSTANCE     = new BeyondCompareReporter();
  static final        String                DIFF_PROGRAM = "/usr/local/bin/bcompare";
  static final        String                MESSAGE      = MessageFormat.format(
      "Unable to find Beyond Compare at {0}", DIFF_PROGRAM);

  public BeyondCompareReporter() {
    super(DIFF_PROGRAM, MESSAGE);
  }
}
