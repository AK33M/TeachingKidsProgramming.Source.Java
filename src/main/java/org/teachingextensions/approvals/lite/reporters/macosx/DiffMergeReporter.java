package org.teachingextensions.approvals.lite.reporters.macosx;

import org.teachingextensions.approvals.lite.reporters.GenericDiffReporter;

import java.text.MessageFormat;

public class DiffMergeReporter extends GenericDiffReporter {
  public static final  DiffMergeReporter INSTANCE     = new DiffMergeReporter();
  private static final String            DIFF_PROGRAM = "/Applications/DiffMerge.app/Contents/MacOS/DiffMerge";
  static final         String            MESSAGE      = MessageFormat.format("Unable to find DiffMerge at {0}",
      DIFF_PROGRAM);

  public DiffMergeReporter() {
    super(DIFF_PROGRAM, GenericDiffReporter.STANDARD_ARGUMENTS, MESSAGE, GenericDiffReporter.TEXT_FILE_EXTENSIONS);
  }
}
