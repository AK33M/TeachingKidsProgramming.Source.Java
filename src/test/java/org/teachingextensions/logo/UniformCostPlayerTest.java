package org.teachingextensions.logo;

import org.junit.Test;
import org.teachingextensions.approvals.lite.reporters.UseReporter;
import org.teachingextensions.approvals.lite.reporters.macosx.BeyondCompareReporter;

@UseReporter(BeyondCompareReporter.class)
public class UniformCostPlayerTest extends PuzzlePlayerTest {
  @Override
  protected PuzzlePlayer getPlayer(Puzzle puzzle) {
    return new UniformCostPlayer(puzzle);
  }

  /**
   * Produces a puzzle solution
   */
  @Test
  public void solve_puzzle() throws Exception {
    verifySolution();
  }

  /**
   * Solve longer puzzle
   */
  @Test
  public void solve_longer_puzzle() throws Exception {
    verifyLongSolution();
  }
}
