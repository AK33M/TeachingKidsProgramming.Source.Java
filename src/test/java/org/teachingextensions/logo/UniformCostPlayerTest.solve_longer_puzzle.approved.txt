From [0, 1, 2, 3, 4, 8, 5, 6, 7] to [0, 1, 2, 3, 4, 5, 6, 7, 8]


{Down = 3}
{Left = -1}
{Left = -1}
{Up = -3}
{Right = 1}
{Down = 3}
{Right = 1}
{Up = -3}
{Left = -1}
{Left = -1}
{Down = 3}
{Right = 1}
{Right = 1}
